base_layer/wallet/src/base_node_service/service.rs

```diff
@@ -158,18 +157,7 @@ where T: WalletBackend + 'static
     }
 
     async fn set_base_node_peer(&mut self, peer: Peer) -> Result<(), BaseNodeServiceError> {
-        let new_state = BaseNodeState {
-            base_node_peer: Some(peer.clone()),
-            ..Default::default()
-        };
-
-        {
-            let mut lock = self.state.write().await;
-            *lock = new_state.clone();
-        }
-        self.wallet_connectivity.set_base_node(peer.node_id.clone()).await?;
-
-        self.publish_event(BaseNodeEvent::BaseNodeStateChanged(new_state));
+        self.wallet_connectivity.set_base_node(peer.clone()).await?;
```

Comment:
This feels like it should happen in the wallet container `set_base_node_peer` function where all the other wallet services have their set_base_node method called.

Reply:
Yeah agree,  will require a bit more of a refactor (events need to be moved around), wanted to get this in quicker.